### ISSUE TITLE
Add `ChoiceMap.switch` for creating choicemaps for switch combinator

### ISSUE THREAD

**Issue by sritchie** [genjax-community/genjax#1313]
I'd like to push common patterns into `ChoiceMap`, or put them somewhere that a user could plausibly create their own sample that they can then go in and change.

Here's a proposal for `switch`:

```python
@staticmethod
def switch(idx: ArrayLike, chms: "list[ChoiceMap]") -> "ChoiceMap":
        acc = ChoiceMap.empty()
        for i, chm in enumerate(chms):
            masked = chm.mask(Flag(jnp.all(i == idx)))
            acc = acc ^ masked
        return acc
```

The choicemaps look like this, with an `XorChm` branch for each one.

```python
XorChm(
  c1=MaskChm(
    c=StaticChm(
      c=ValueChm(v=<jax.Array(-1.2515389, dtype=float32)>),
      addr='x1',
    ),
    flag=Flag(f=<jax.Array(True, dtype=bool)>),
  ),
  c2=MaskChm(
    c=StaticChm(
      c=ValueChm(v=<jax.Array(0, dtype=int32)>),
      addr='x2',
    ),
    flag=Flag(f=<jax.Array(False, dtype=bool)>),
  ),
)
```

**Comment by sritchie** [comment]
@femtomc notes that this is not how you'd do this, you wouldn't want to build something with all the masks.
